### ISSUE TITLE
phashion: fatal error: 'cstdio' file not found

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1637]
`phashion`gem is one of dependencies of `image_processing` gem. There is no issue to install this gem in MRI.

To reproduce:

```
gem install phashion
```

Error:
```
Building native extensions. This could take a while...
ERROR:  Error installing phashion:
	ERROR: Failed to build gem native extension.

    current directory: ~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext
~/.rubies/truffleruby-1.0.0-rc14/bin/truffleruby -I ~/.rubies/truffleruby-1.0.0-rc14/lib/mri -r ./siteconf20190324-4723-4553dt.rb extconf.rb
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking for g++... g++
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking for style of include used by make... GNU
checking dependency style of g++... gcc3
checking for gcc... gcc
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... unsupported
checking dependency style of gcc... gcc3
checking for /etc/gentoo-release... no
checking for /etc/redhat-release... no
checking for /etc/slackware-version... no
checking for /etc/debian_version... no
checking for /etc/SuSErelease... no
checking build system type... i386-apple-darwin18.2.0
checking host system type... i386-apple-darwin18.2.0
checking for the pthreads library -lpthreads... no
checking whether pthreads work without any flags... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking if more special flags are required for pthreads... -D_THREAD_SAFE
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking whether ln -s works... yes
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /Library/Developer/CommandLineTools/usr/bin/ld
checking if the linker (/Library/Developer/CommandLineTools/usr/bin/ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking the maximum length of command line arguments... 196608
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert i386-apple-darwin18.2.0 file names to i386-apple-darwin18.2.0 format... func_convert_file_noop
checking how to convert i386-apple-darwin18.2.0 file names to toolchain format... func_convert_file_noop
checking for /Library/Developer/CommandLineTools/usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... no
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for mt... no
checking if : is a manifest tool... no
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking for -force_load linker flag... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... yes
checking for gcc option to produce PIC... -fno-common -DPIC
checking if gcc PIC flag -fno-common -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/Library/Developer/CommandLineTools/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin18.2.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /Library/Developer/CommandLineTools/usr/bin/ld
checking if the linker (/Library/Developer/CommandLineTools/usr/bin/ld) is GNU ld... no
checking whether the g++ linker (/Library/Developer/CommandLineTools/usr/bin/ld) supports shared libraries... yes
checking for g++ option to produce PIC... -fno-common -DPIC
checking if g++ PIC flag -fno-common -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/Library/Developer/CommandLineTools/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin18.2.0 dyld
checking how to hardcode library paths into programs... immediate

*** Configuring image hash ***

checking CImg.h usability... no
checking CImg.h presence... no
checking for CImg.h... no
checking whether CImg.h is in the current or src directory.... yes
checking for png_create_read_struct in -lpng... yes
checking for jpeg_read_header in -ljpeg... yes
checking for fftw_plan_dft_r2c_1d in -lfftw3... yes
checking for size_t... yes
checking for uint32_t... yes
checking for uint8_t... yes
checking for uint64_t... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating examples/Makefile
config.status: creating pHash.pc
config.status: creating bindings/Makefile
config.status: creating bindings/java/Makefile
config.status: creating config.h
config.status: creating pHash-config.h
config.status: executing depfiles commands
config.status: executing libtool commands
/Library/Developer/CommandLineTools/usr/bin/make  all-recursive
Making all in src
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I..   -D_THREAD_SAFE -I/usr/local/include -Dcimg_use_png -Dcimg_use_jpeg -Dcimg_use_fftw3  -pthread -ffast-math -O3 -MT pHash.lo -MD -MP -MF .deps/pHash.Tpo -c -o pHash.lo pHash.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -D_THREAD_SAFE -I/usr/local/include -Dcimg_use_png -Dcimg_use_jpeg -Dcimg_use_fftw3 -pthread -ffast-math -O3 -MT pHash.lo -MD -MP -MF .deps/pHash.Tpo -c pHash.cpp  -fno-common -DPIC -o pHash.o
In file included from pHash.cpp:25:
In file included from ./pHash.h:50:
../CImg.h:2160:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static char min() { return (char)(-1L<<(8*sizeof(char)-1)); }
                                        ~~~^
../CImg.h:2161:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static char max() { return ~((char)(-1L<<(8*sizeof(char)-1))); }
                                          ~~~^
../CImg.h:2171:58: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static signed char min() { return (signed char)(-1L<<(8*sizeof(signed char)-1)); }
                                                      ~~~^
../CImg.h:2172:60: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static signed char max() { return ~((signed char)(-1L<<(8*sizeof(signed char)-1))); }
                                                        ~~~^
../CImg.h:2193:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static short min() { return (short)(-1L<<(8*sizeof(short)-1)); }
                                          ~~~^
../CImg.h:2194:48: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static short max() { return ~((short)(-1L<<(8*sizeof(short)-1))); }
                                            ~~~^
../CImg.h:2215:42: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static int min() { return (int)(-1L<<(8*sizeof(int)-1)); }
                                      ~~~^
../CImg.h:2216:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static int max() { return ~((int)(-1L<<(8*sizeof(int)-1))); }
                                        ~~~^
../CImg.h:2237:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static long min() { return (long)(-1L<<(8*sizeof(long)-1)); }
                                        ~~~^
../CImg.h:2238:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static long max() { return ~((long)(-1L<<(8*sizeof(long)-1))); }
                                          ~~~^
pHash.cpp:415:1: warning: control reaches end of non-void function [-Wreturn-type]
}
^
11 warnings generated.
mv -f .deps/pHash.Tpo .deps/pHash.Plo
/bin/sh ../libtool --tag=CXX   --mode=link g++  -pthread -ffast-math -O3 -no-undefined -L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib -L/usr/local/lib -L/usr/local/lib -o libpHash.la -rpath ~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib pHash.lo    -lpng -ljpeg -lfftw3
libtool: link: ar cru .libs/libpHash.a  pHash.o
libtool: link: ranlib .libs/libpHash.a
libtool: link: ( cd ".libs" && rm -f "libpHash.la" && ln -s "../libpHash.la" "libpHash.la" )
Making all in bindings
make[3]: Nothing to be done for `all-am'.
Making all in examples
g++ -DHAVE_CONFIG_H -I. -I.. -I../src  -D_THREAD_SAFE -I/usr/local/include -Dcimg_use_png -Dcimg_use_jpeg -Dcimg_use_fftw3  -pthread -ffast-math -O3 -MT test_texthash.o -MD -MP -MF .deps/test_texthash.Tpo -c -o test_texthash.o test_texthash.cpp
In file included from test_texthash.cpp:28:
In file included from ../src/pHash.h:50:
../CImg.h:2160:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static char min() { return (char)(-1L<<(8*sizeof(char)-1)); }
                                        ~~~^
../CImg.h:2161:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static char max() { return ~((char)(-1L<<(8*sizeof(char)-1))); }
                                          ~~~^
../CImg.h:2171:58: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static signed char min() { return (signed char)(-1L<<(8*sizeof(signed char)-1)); }
                                                      ~~~^
../CImg.h:2172:60: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static signed char max() { return ~((signed char)(-1L<<(8*sizeof(signed char)-1))); }
                                                        ~~~^
../CImg.h:2193:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static short min() { return (short)(-1L<<(8*sizeof(short)-1)); }
                                          ~~~^
../CImg.h:2194:48: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static short max() { return ~((short)(-1L<<(8*sizeof(short)-1))); }
                                            ~~~^
../CImg.h:2215:42: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static int min() { return (int)(-1L<<(8*sizeof(int)-1)); }
                                      ~~~^
../CImg.h:2216:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static int max() { return ~((int)(-1L<<(8*sizeof(int)-1))); }
                                        ~~~^
../CImg.h:2237:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static long min() { return (long)(-1L<<(8*sizeof(long)-1)); }
                                        ~~~^
../CImg.h:2238:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static long max() { return ~((long)(-1L<<(8*sizeof(long)-1))); }
                                          ~~~^
test_texthash.cpp:68:26: warning: format specifies type 'intmax_t' (aka 'long') but the argument has type 'off_t' (aka 'long long') [-Wformat]
        printf(" %jd %jd %d\n", matches[j].first_index, matches[j].second_index,matches[j].length);
                 ~~~            ^~~~~~~~~~~~~~~~~~~~~~
                 %lld
test_texthash.cpp:68:50: warning: format specifies type 'intmax_t' (aka 'long') but the argument has type 'off_t' (aka 'long long') [-Wformat]
        printf(" %jd %jd %d\n", matches[j].first_index, matches[j].second_index,matches[j].length);
                     ~~~                                ^~~~~~~~~~~~~~~~~~~~~~~
                     %lld
12 warnings generated.
mv -f .deps/test_texthash.Tpo .deps/test_texthash.Po
/bin/sh ../libtool --tag=CXX   --mode=link g++  -pthread -ffast-math -O3  -L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib -L/usr/local/lib -L/usr/local/lib -o test_texthash test_texthash.o ../src/libpHash.la -lpng -ljpeg -lfftw3
libtool: link: g++ -pthread -ffast-math -O3 -o test_texthash test_texthash.o -Wl,-bind_at_load  -L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib -L/usr/local/lib ../src/.libs/libpHash.a -lpng -ljpeg -lfftw3 -pthread
ld: warning: directory not found for option '-L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib'
g++ -DHAVE_CONFIG_H -I. -I.. -I../src  -D_THREAD_SAFE -I/usr/local/include -Dcimg_use_png -Dcimg_use_jpeg -Dcimg_use_fftw3  -pthread -ffast-math -O3 -MT test_texthash2.o -MD -MP -MF .deps/test_texthash2.Tpo -c -o test_texthash2.o test_texthash2.cpp
In file included from test_texthash2.cpp:27:
In file included from ../src/pHash.h:50:
../CImg.h:2160:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static char min() { return (char)(-1L<<(8*sizeof(char)-1)); }
                                        ~~~^
../CImg.h:2161:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static char max() { return ~((char)(-1L<<(8*sizeof(char)-1))); }
                                          ~~~^
../CImg.h:2171:58: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static signed char min() { return (signed char)(-1L<<(8*sizeof(signed char)-1)); }
                                                      ~~~^
../CImg.h:2172:60: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static signed char max() { return ~((signed char)(-1L<<(8*sizeof(signed char)-1))); }
                                                        ~~~^
../CImg.h:2193:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static short min() { return (short)(-1L<<(8*sizeof(short)-1)); }
                                          ~~~^
../CImg.h:2194:48: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static short max() { return ~((short)(-1L<<(8*sizeof(short)-1))); }
                                            ~~~^
../CImg.h:2215:42: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static int min() { return (int)(-1L<<(8*sizeof(int)-1)); }
                                      ~~~^
../CImg.h:2216:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static int max() { return ~((int)(-1L<<(8*sizeof(int)-1))); }
                                        ~~~^
../CImg.h:2237:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static long min() { return (long)(-1L<<(8*sizeof(long)-1)); }
                                        ~~~^
../CImg.h:2238:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static long max() { return ~((long)(-1L<<(8*sizeof(long)-1))); }
                                          ~~~^
10 warnings generated.
mv -f .deps/test_texthash2.Tpo .deps/test_texthash2.Po
/bin/sh ../libtool --tag=CXX   --mode=link g++  -pthread -ffast-math -O3  -L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib -L/usr/local/lib -L/usr/local/lib -o test_texthash2 test_texthash2.o ../src/libpHash.la -lpng -ljpeg -lfftw3
libtool: link: g++ -pthread -ffast-math -O3 -o test_texthash2 test_texthash2.o -Wl,-bind_at_load  -L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib -L/usr/local/lib ../src/.libs/libpHash.a -lpng -ljpeg -lfftw3 -pthread
ld: warning: directory not found for option '-L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib'
g++ -DHAVE_CONFIG_H -I. -I.. -I../src  -D_THREAD_SAFE -I/usr/local/include -Dcimg_use_png -Dcimg_use_jpeg -Dcimg_use_fftw3  -pthread -ffast-math -O3 -MT test_imagephash.o -MD -MP -MF .deps/test_imagephash.Tpo -c -o test_imagephash.o test_imagephash.cpp
In file included from test_imagephash.cpp:30:
In file included from ../src/pHash.h:50:
../CImg.h:2160:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static char min() { return (char)(-1L<<(8*sizeof(char)-1)); }
                                        ~~~^
../CImg.h:2161:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static char max() { return ~((char)(-1L<<(8*sizeof(char)-1))); }
                                          ~~~^
../CImg.h:2171:58: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static signed char min() { return (signed char)(-1L<<(8*sizeof(signed char)-1)); }
                                                      ~~~^
../CImg.h:2172:60: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static signed char max() { return ~((signed char)(-1L<<(8*sizeof(signed char)-1))); }
                                                        ~~~^
../CImg.h:2193:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static short min() { return (short)(-1L<<(8*sizeof(short)-1)); }
                                          ~~~^
../CImg.h:2194:48: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static short max() { return ~((short)(-1L<<(8*sizeof(short)-1))); }
                                            ~~~^
../CImg.h:2215:42: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static int min() { return (int)(-1L<<(8*sizeof(int)-1)); }
                                      ~~~^
../CImg.h:2216:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static int max() { return ~((int)(-1L<<(8*sizeof(int)-1))); }
                                        ~~~^
../CImg.h:2237:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static long min() { return (long)(-1L<<(8*sizeof(long)-1)); }
                                        ~~~^
../CImg.h:2238:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static long max() { return ~((long)(-1L<<(8*sizeof(long)-1))); }
                                          ~~~^
10 warnings generated.
mv -f .deps/test_imagephash.Tpo .deps/test_imagephash.Po
/bin/sh ../libtool --tag=CXX   --mode=link g++  -pthread -ffast-math -O3  -L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib -L/usr/local/lib -L/usr/local/lib -o test_image test_imagephash.o ../src/libpHash.la -lpng -ljpeg -lfftw3
libtool: link: g++ -pthread -ffast-math -O3 -o test_image test_imagephash.o -Wl,-bind_at_load  -L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib -L/usr/local/lib ../src/.libs/libpHash.a -lpng -ljpeg -lfftw3 -pthread
ld: warning: directory not found for option '-L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib'
g++ -DHAVE_CONFIG_H -I. -I.. -I../src  -D_THREAD_SAFE -I/usr/local/include -Dcimg_use_png -Dcimg_use_jpeg -Dcimg_use_fftw3  -pthread -ffast-math -O3 -MT test_mhimagehash.o -MD -MP -MF .deps/test_mhimagehash.Tpo -c -o test_mhimagehash.o test_mhimagehash.cpp
In file included from test_mhimagehash.cpp:4:
In file included from ../src/pHash.h:50:
../CImg.h:2160:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static char min() { return (char)(-1L<<(8*sizeof(char)-1)); }
                                        ~~~^
../CImg.h:2161:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static char max() { return ~((char)(-1L<<(8*sizeof(char)-1))); }
                                          ~~~^
../CImg.h:2171:58: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static signed char min() { return (signed char)(-1L<<(8*sizeof(signed char)-1)); }
                                                      ~~~^
../CImg.h:2172:60: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static signed char max() { return ~((signed char)(-1L<<(8*sizeof(signed char)-1))); }
                                                        ~~~^
../CImg.h:2193:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static short min() { return (short)(-1L<<(8*sizeof(short)-1)); }
                                          ~~~^
../CImg.h:2194:48: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static short max() { return ~((short)(-1L<<(8*sizeof(short)-1))); }
                                            ~~~^
../CImg.h:2215:42: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static int min() { return (int)(-1L<<(8*sizeof(int)-1)); }
                                      ~~~^
../CImg.h:2216:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static int max() { return ~((int)(-1L<<(8*sizeof(int)-1))); }
                                        ~~~^
../CImg.h:2237:44: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static long min() { return (long)(-1L<<(8*sizeof(long)-1)); }
                                        ~~~^
../CImg.h:2238:46: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
      static long max() { return ~((long)(-1L<<(8*sizeof(long)-1))); }
                                          ~~~^
10 warnings generated.
mv -f .deps/test_mhimagehash.Tpo .deps/test_mhimagehash.Po
/bin/sh ../libtool --tag=CXX   --mode=link g++  -pthread -ffast-math -O3  -L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib -L/usr/local/lib -L/usr/local/lib -o test_mhimagehash test_mhimagehash.o ../src/libpHash.la -lpng -ljpeg -lfftw3
libtool: link: g++ -pthread -ffast-math -O3 -o test_mhimagehash test_mhimagehash.o -Wl,-bind_at_load  -L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib -L/usr/local/lib ../src/.libs/libpHash.a -lpng -ljpeg -lfftw3 -pthread
ld: warning: directory not found for option '-L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib'
make[2]: Nothing to be done for `all-am'.
Making install in src
 .././install-sh -c -d '~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib'
 /bin/sh ../libtool   --mode=install /usr/bin/install -c   libpHash.la '~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib'
libtool: install: /usr/bin/install -c .libs/libpHash.lai ~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib/libpHash.la
libtool: install: /usr/bin/install -c .libs/libpHash.a ~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib/libpHash.a
libtool: install: chmod 644 ~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib/libpHash.a
libtool: install: ranlib ~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib/libpHash.a
 .././install-sh -c -d '~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/include'
 /usr/bin/install -c -m 644 pHash.h '~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/include'
Making install in bindings
make[3]: Nothing to be done for `install-exec-am'.
make[3]: Nothing to be done for `install-data-am'.
Making install in examples
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
make[2]: Nothing to be done for `install-exec-am'.
 ./install-sh -c -d '~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/include'
 /usr/bin/install -c -m 644 pHash-config.h '~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/include'
 ./install-sh -c -d '~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib/pkgconfig'
 /usr/bin/install -c -m 644 pHash.pc '~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib/pkgconfig'
tar xzf ~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/pHash-0.9.6.tar.gz 2>&1
env CXXFLAGS=' -pthread' CFLAGS=' -I~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/include  -L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib -L/usr/local/lib  -x c++ ' LDFLAGS=' -L~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/lib -L/usr/local/lib ' ./configure --prefix=~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext --disable-audio-hash --disable-video-hash --disable-shared --with-pic 2>&1
make || true 2>&1
make install || true 2>&1
mv CImg.h ../include 2>&1
checking for sqlite3ext.h... no
creating Makefile

current directory: ~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext
make "DESTDIR=" clean

current directory: ~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext
make "DESTDIR="
compiling phashion_ext.c
In file included from phashion_ext.c:2:
In file included from ~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/include/pHash.h:50:
~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0/ext/phashion_ext/include/CImg.h:72:10: fatal error: 'cstdio' file not found
#include <cstdio>
         ^~~~~~~~
1 error generated.
make: *** [phashion_ext.bc] Error 1

make failed, exit code 2

Gem files will remain installed in ~/.gem/truffleruby/2.6.2/gems/phashion-1.2.0 for inspection.
Results logged to ~/.gem/truffleruby/2.6.2/extensions/x86_64-darwin/2.6.0/phashion-1.2.0/gem_make.out

```

**Comment by eregon** [comment]
Thanks for the report.
I had a quick look, and `phashion` tries to compile a C++ extension with a C command line for compiling, and tweaking the CFLAGS. That's obviously not recommended and going to lead to many problems.

One small issue we can fix is moving `-xc` of `RbConfig::MAKEFILE_CONFIG['COMPILE_C']` earlier so it can still be overridden by `CFLAGS` which `phashion` modifies (notably adding `-xc++`). That lets the extension `.c` compile, but doesn't link:
```
/usr/bin/../lib/gcc/x86_64-redhat-linux/7/../../../../lib64/crt1.o: In function `_start':
(.text+0x20): undefined reference to `main'
/tmp/--07c5cd.o: In function `Init_phashion_ext':
-:(.text+0x9): undefined reference to `rb_tr_get_Object'
-:(.text+0x24): undefined reference to `rb_intern'
-:(.text+0x30): undefined reference to `rb_const_get'
-:(.text+0x61): undefined reference to `rb_define_singleton_method'
...
```
Which is probably because of the `CFLAGS` tweak above and/or `$LIBS = "-lstdc++"`.
I think in this case, it should be fixed upstream to use the existing mechanism to compile a Ruby C++ extension.

**Comment by eregon** [comment]
I tried with the new Sulong toolchain and `phashion` installs successfully now:
```
$ gem i phashion                
Building native extensions. This could take a while...
Successfully installed phashion-1.2.0
1 gem installed
```
So this should be fixed in 19.3.0.
Thank you for the report.
